src/hooks/useConnectToValidChain.tsx

```diff
@@ -0,0 +1,31 @@
+import { useNetwork } from "wagmi";
+
+/**
+ * @remarks if user is not connected to supported chains (Mainnet 1, or Goerli 5), this hook will prompt user to switch to Mainnet
+ * @returns void
+ */
+export const useConnectToValidChain = async () => {
```

Comment:
This function _works_, but is not quite the "react way". 

Switching chains is a side effect, which should hint that it should be in a `useEffect`. Basically anything with an async request should be.

The component that uses this hook could re-render many times in a short space of time (especially on an initial load with multiple data fetches that each set a state). In the event that the switch chain request was slower to resolve than the rerenders, it'd be firing off a lot of requests.

TLDR, wrap everything from line 10 onwards in a `useEffect`.

Reply:
I switched everything with the wagmi hook `useSwitchNetwork` and removed the other sloppy errors. Now it's more the React way